RELEASE_CHECKLIST.md

```diff
@@ -6,11 +6,14 @@
 - Copy the contents of the CHANGELOG for this release into commit message,
   using `git commit -s --cleanup=whitespace` so the that the lines beginning
   with `#` are not removed.
+- Run `cargo publish --dry-run` in the `solang-parser` folder.
+- Publish the solang-parser crate, by running `cargo publish` in the solang-parser folder
```

Comment:
Do not publish anything until the release process is done

Reply:
We have a chicken and egg problem here. What if you publish solang-parser before the release PR is merged, and then further changes are needed in solang-parser that come out in review, or other PRs that get merged in the mean time change the solang-parser crate?

Reply:
How about moving the publishing of solang-parser to the moment before the release PR is merged, after review?

Reply:
We also need an additional step in the the check-list: when changing llvm version, the CI image has to be re-generated using Github Actions manual trigger.

Reply:
Fixed